spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -159,16 +181,152 @@ public ListenableFuture<Job> writeDataToTable(
   }
 
   /**
-   * @return the name of the BigQuery dataset that the template is operating in.
+   * This method uses BigQuery Storage Write API to write new line delimited JSON file to the
+   * specified table. This method creates a table with the specified schema.
+   *
+   * @param tableName name of the table to write to
+   * @param jsonInputStream input stream of the json file to be written
+   * @return {@link ListenableFuture} containing the WriteApiResponse indicating completion of
+   *     operation
    */
+  @Override
+  public ListenableFuture<WriteApiResponse> writeJsonStream(
+      String tableName, InputStream jsonInputStream, Schema schema) {
+    createTable(tableName, schema); // create table if it's not already created
+    return writeJsonStream(tableName, jsonInputStream);
+  }
+
+  private void createTable(
+      String tableName, Schema schema) { // create table if it's not already created
+    TableId tableId = TableId.of(datasetName, tableName);
+    Table table = bigQuery.getTable(TableId.of(datasetName, tableName));
+    if (table == null || !table.exists()) {
+      TableDefinition tableDefinition = StandardTableDefinition.of(schema);
+      TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefinition).build();
+      bigQuery.create(tableInfo);
+    }
+  }
+
+  /**
+   * This method uses BigQuery Storage Write API to write new line delimited JSON file to the
+   * specified table. The Table should already be created as BigQuery Storage Write API doesn't
+   * create it automatically.
+   *
+   * @param tableName name of the table to write to
+   * @param jsonInputStream input stream of the json file to be written
+   * @return {@link ListenableFuture} containing the WriteApiResponse indicating completion of
+   *     operation
+   */
+  @Override
+  public ListenableFuture<WriteApiResponse> writeJsonStream(
+      String tableName, InputStream jsonInputStream) {
+
+    SettableListenableFuture<WriteApiResponse> writeApiFutureResponse =
+        new SettableListenableFuture<>();
+    CompletableFuture<Void> completableFuture =
+        CompletableFuture.runAsync(
```

Comment:
Non-actionable, just a note for future -- we may want to allow users to inject a custom threadpool here, like we do for Pub/Sub. Although with BigQuery writing, I doubt there is as high of parallelism as with Pub/Sub.

Reply:
Good idea, captured it as a feedback @  go/bq-write-api-sc-gcp-integration